src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenForEachTests.cs

```diff
@@ -1736,11 +1737,18 @@ class Enumerable1
     public bool MoveNext() { return ++x < 4; }
     public void Dispose() { System.Console.WriteLine(""Done with DisposableEnumerator""); }
 }";
-            // ILVerify: Return type is ByRef, TypedReference, ArgHandle, or ArgIterator.
-            var compilation = CompileAndVerify(source, parseOptions: TestOptions.Regular7_3, verify: Verification.FailsILVerify, expectedOutput: @"
-1
-2
-3");
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular7_3);
+            comp.VerifyDiagnostics(
+                // error CS8370: Feature 'using declarations' is not available in C# 7.3. Please use language version 8.0 or greater.
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3).WithArguments("using declarations", "8.0").WithLocation(1, 1)
```

Comment:
>"using declarations" [](http://example.com/codeflow?start=88&length=20)

This seems incorrect since there are no using declarations.

Reply:
The "using declarations" feature bundled two things: using declarations and pattern-based disposal.
I'm fine to split but it snowballs the PR a bit.

Reply:
I don't think we need to separate "pattern-based disposal" from "using declarations", but it feels like this error message should reference "async streams" rather than "using declarations".

Reply:
You can hit this with `foreach` on a ref struct. It's not specific to async streams...